tests/framework/Models/External/serialize_pyomo.xml

```diff
@@ -0,0 +1,111 @@
+<?xml version="1.0" ?>
+<Simulation>
+  <TestInfo>
+    <name>framework.Models.External.serialize_pyomo</name>
+    <author>cogljj</author>
+    <created>2021-08-18</created>
+    <classesTested>Models.ROM</classesTested>
+    <description>
+      This test serializes a ROM to use with PYOMO.
+      It can be used with:
+      cd tests/framework/Models/External
+      python3 SerializePyomo/rom_out.py -r SerializePyomo/rom_pickle.pk -f ../../../..
+    </description>
+    <analytic>
+      This test uses the analytic model ''attenuate``, which is documented in the analytical test documentation.
+      Additionally, values of the ''from`` variables are exactly determined because they are set in the extmod methods:
+      \begin{itemize}
+        \item fromReadMoreXML: pi         (3.14159)
+        \item fromInit       : sqrt(pi)   (1.77245)
+        \item fromCNISelf    : pi/2       (1.57080)
+        \item fromCNIDict    : 2*sqrt(pi) (3.54491)
+      \end{itemize}
+      The exit strength ''ans`` is analytic with results as follows (Note that for testing purposes, the model always adds
+      0.05 to \texttt{y2} before calculating the exit strength):
+      \begin{itemize}
+        \item \texttt{y1}, \texttt{y2}, \texttt{ans}
+        \item 0.0, 0.0, 0.97531
+        \item 0.0, 1.0, 0.59156
+        \item 1.0, 0.0, 0.59156
+        \item 1.0, 1.0, 0.35880
+      \end{itemize}
+    </analytic>
+  </TestInfo>
+
+  <RunInfo>
+    <WorkingDir>SerializePyomo</WorkingDir>
+    <Sequence>sample, train, serialize, pickle</Sequence>
+  </RunInfo>
+
+  <Files>
+    <Input name="rom_out.py" type="">rom_out.py</Input>
+    <Input name="rom_pickle.pk" type="">rom_pickle.pk</Input>
+  </Files>
+
+  <Steps>
+    <MultiRun name="sample">
+      <Input class="DataObjects" type="PointSet">dummyIN</Input>
+      <Model class="Models" type="ExternalModel">attenuate</Model>
+      <Sampler class="Samplers" type="Grid">grid</Sampler>
+      <Output class="DataObjects" type="PointSet">samples</Output>
+    </MultiRun>
+    <RomTrainer name="train">
+      <Input class="DataObjects" type="PointSet">samples</Input>
+      <Output class="Models" type="ROM">out_rom</Output>
+    </RomTrainer>
+    <IOStep name="serialize">
+      <Input class="Models" type="ROM">out_rom</Input>
+      <Output class="Files" type="">rom_out.py</Output>
+    </IOStep>
+    <IOStep name="pickle">
+      <Input class="Models" type="ROM">out_rom</Input>
+      <Output class="Files" type="">rom_pickle.pk</Output>
```

Comment:
@aalfonsi @joshua-cogliati-inl  should we use the type attribute to indicate the serialization and pickle process?

Reply:
Using the type attribute sounds reasonable.

Reply:
I changed this with 313f69a47 to use the type attribute in the File node.